clang/lib/Headers/cheriintrin.h

```diff
@@ -67,7 +67,10 @@ typedef long cheri_otype_t;
 #define cheri_type_copy(x, y) __builtin_cheri_cap_type_copy((x), (y))
 
 /* Capability permissions: */
-typedef enum __attribute__((flag_enum, enum_extensibility(open))) {
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wfixed-enum-extension"
+/* Use __PTRADDR_TYPE__ as the underlying type to avoid unnecessary truncates */
+typedef enum __attribute__((flag_enum, enum_extensibility(open))) : __PTRADDR_TYPE__ {
```

Comment:
What is the point of this enum? Are we not better off with a list of `#define CHERI_PERM_FOO (__CHERI_CAP_PERMISSION_FOO__)`? We can make sure the preprocessor gives the latter the right type (or cast each one here, but that seems worse).

Reply:
Good point, I can't remember why I originally made this an enum. Maybe to avoid redefinition warnings in CheriBSD?

Reply:
That only works so long as cheriintrin.h is included _before_ cherireg.h, which is the opposite of what style(9) says

Reply:
It could also be that the reason was for nicer visualization in debuggers, but I'm not sure if that actually works.